### PR TITLE
fix(provider/oracle): Add OracleImageFinder for findImageByTags to work

### PR DESCRIPTION
**Pull request by desagar**
Added an OracleImageFinder implementation which is needed for the Oracle cloud's find image pipeline stage to work correctly.

Related PRs:
https://github.com/spinnaker/deck/pull/5605
https://github.com/spinnaker/clouddriver/pull/2874

@markxnelson

### CODE REVIEW

**Comment by shihchang** [comment]
@spinnaker/reviewers

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+@Component
+public class OracleImageFinder implements ImageFinder {
+  private static final Logger log = LoggerFactory.getLogger(OracleImageFinder.class);
+
+  @Autowired
+  OortService oortService;
+
+  @Autowired
+  ObjectMapper objectMapper;
```
Comment (robzienert) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy]: Use autowired constructors over properties.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (robzienert) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy]: Please avoid adding more Groovy classes in main source tree whenever possible.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                                         Map<String, String> freeformTags) {
+
+    StageData stageData = (StageData) stage.mapTo(StageData.class);
+    List<OracleImage> allMatchedImages = oortService
+      .findImage(getCloudProvider(), packageName, null, null, prefixTags(freeformTags))
+      .collect { objectMapper.convertValue(it, OracleImage.class) }
+
+    //For each region, find the matching image for that region. If there are multiple matches, just
+    //use the first one. Note: sort order of images is not defined by the OCI SDK.
+    //TODO - add created date and look for most recent, similar to Amazon.
```
Comment (robzienert) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy]: I'd tend to think that if this stage isn't returning the most recent image, the logic is fundamentally broken as the current functionality wouldn't match end-users' expectations.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy (hunk 2 above)
Comment (desagar) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy]: Is Java preferred?

Comment (robzienert) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy]: Yes

**Pull request merged**